src/plotting/animations.jl

```diff
@@ -1,109 +1,113 @@
-  * `framerate = 5` : Rate of drawing new collisions (per second).
-    Both during showing animation and during saving it.
-  * `col_to_plot = 5` : How many previous collisions are shown during the animation.
-  * `particle_kwargs` : A `NamedTuple` of keywords passed to [`plot`](@ref).
-  * `orbit_kwargs` : A `NamedTuple` of keywrods passed to `PyPlot.plot`
-    which plots the orbit of the particle (`line` object).
-  * `savename = nothing` : If `nothing`, nothing is saved.
-    If you give a string instead (filename *including path*) an animation `.mp4`
-    will be saved with name `savename.mp4`. **This requires
-    [`ffmpeg`](https://www.ffmpeg.org) to be accessible from the command line.**
-  * `deletefigs = true` : When producing the animation, each frame is saved
-    (in the same folder) but then deleted after the animation is created. You
-    can choose to keep them by passing `false`.
-  * `figsize = (7.2, 7.2)` : `PyPlot` figure size.
-  * `dpi = 100` : DPI of saved figures *and* resulting animation.
-
-### Return
-The function returns `a, b, c`. Do `a[:remove](); b[:remove](); c[:remove]()` to clear
-the particle out of the figure.
+  * `dt = 0.01` : Time resolution used for production of time series(see 
+    [`timeseries`](@ref). It is not recommended to increase this value to 
+    preserve the accuracy of the plots.
+  * `frameskip = 5` : The amount of `dt`-steps in between succesive frames
+  * `tailtime = 1.0` : The length of the "tail" trailing the particle in time 
+    units
+  * `colors` : An array of valid Matplotlib colors for the "tails". If `colors`
+    is shorter than `ps`, colors are reused. Defaults to the standard      
+    Matplotlib color sequence
+  * `particle_kwargs::NamedTuple` : Additional keyword arguments passed to the
+    `plot` function for particles.
+  * `tail_kwargs::NamedTuple`: Additional keyword arguments passed to the `plot` 
+    function for "tails".
 """
-function animate_evolution(par::AbstractParticle, bd, colnumber, raysplit = nothing;
-    framerate = 5, col_to_plot = 5, savename = nothing,
-    particle_kwargs = NamedTuple(), orbit_kwargs = NamedTuple(), newfig = true,
-    disable_axis = false, deletefigs = true, dpi = 100,
-    figsize = (7.2, 7.2))
 
-    p = copy(par)
-    if newfig == true
-        fig = PyPlot.figure(figsize = figsize)
-        plot(bd; ax = PyPlot.gca())
+function animate_evolution(ps::AbstractVector{<:AbstractParticle{T}},
+                           bd::Billiard, t, raysplitters = nothing;
+                           dt  = 0.01, frameskip = 5, tailtime = 1.0,
+                           colors = nothing, particle_kwargs = NamedTuple(),
+                           tail_kwargs = NamedTuple()) where {T}
+
+    nps = length(ps)
+    taillength = round(Int, tailtime/dt)
+
+    ax = PyPlot.gca()
+    plot(bd, ax = ax)
+    
+    plotframes = Vector{Function}(undef, nps)
+    skipframes = Vector{Function}(undef, nps)
```

Comment:
That is okay for now, but not a good long-term solution, as this object is highly unstable.

A better design is to save the line structs in arrays and define genereic functions that take xt,yt etc.

Reply:
But don't do anything about it, leave it as is for now since it "works".